internal/p/p.go

```diff
@@ -20,20 +19,30 @@ func NewCommand() *cobra.Command {
 			if err != nil {
 				return err
 			}
-
 			query := strings.Join(args, " ")
+			queryType := "track"
 
 			deviceID, err := cmd.Flags().GetString("device-id")
 			if err != nil {
 				return err
 			}
 
-			playlistName, err := cmd.Flags().GetString("playlist")
+			contextQuery, err := cmd.Flags().GetString("playlist")
 			if err != nil {
 				return err
 			}
 
-			status, err := p(api, query, playlistName, deviceID)
+			if contextQuery == "" {
+				contextQuery, err = cmd.Flags().GetString("album")
+				if err != nil {
+					return err
+				}
+				queryType = "album"
+			} else {
+				queryType = "playlist"
+			}
```

Comment:
This logic looks identical to the logic in the `Play()` command. Can we call the `Play()` command from inside `p()` to avoid writing this logic twice?